lix/packages/sdk/src/database/initDb.test.ts

```diff
@@ -73,3 +73,26 @@ test("files should be able to have metadata", async () => {
 
 	expect(updatedFile.metadata?.primary_key).toBe("something-else");
 });
+
+test("change edges can't reference themselves", async () => {
+	const sqlite = await createInMemoryDatabase({
+		readOnly: false,
+	});
+	const db = initDb({ sqlite });
+
+	try {
+		await db
+			.insertInto("change_edge")
+			.values({
+				parent_id: "change1",
+				child_id: "change1",
+			})
+			.returningAll()
+			.execute();
+	} catch (error) {
+		// the sqite3 error is not exposed
+		expect((error as any).name).toBe("SQLite3Error");
```

Comment:
Didn't thought of using normal try catch in favour of vitests expect to throw... BUT - this tests passes if the insert doesn't fail? Consider declaring an undefined error before the try, assign the error within the catch and expect after the catch

Reply:
Good one